core/src/test/java/com/cosium/spring/data/jpa/entity/graph/repository/StreamingRepositoryTest.java

```diff
@@ -22,13 +26,28 @@ class StreamingRepositoryTest extends BaseTest {
   @Inject private MyService myService;
 
   @Test
-  void test() {
-    assertThat(myService.list("Product 1")).isNotEmpty();
+  @DisplayName("Given noop eg when finding products then brand should not be initialized")
+  void test1() {
+    List<Product> products = myService.list("Product 1", EntityGraph.NOOP);
+    assertThat(products).isNotEmpty();
+    for (Product product : products) {
+      assertThat(Hibernate.isInitialized(product.getBrand())).isFalse();
+    }
+  }
+
+  @Test
+  @DisplayName("Given brand eg when finding products then brand should be initialize")
+  void test2() {
+    List<Product> products = myService.list("Product 1", NamedEntityGraph.loading(Product.BRAND_EG));
+    assertThat(products).isNotEmpty();
+    for (Product product : products) {
+      assertThat(Hibernate.isInitialized(product.getBrand())).isTrue();
+    }
   }
 
   interface ProductRepository extends EntityGraphRepository<Product, Long> {
 
-    Stream<Product> findByName(String name);
+    Stream<Product> findByName(String name, EntityGraph entityGraph);
```

Comment:
Existing repository methods should no be changed because existing tests rely on them.

Reply:
ok, reverted this change + added dedicated method accepting `EntityGraph` parameter